src/gitops/components/appset/generators/GitGenerator.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { DescriptionList, DescriptionListDescription, DescriptionListGroup, DescriptionListTerm } from '@patternfly/react-core';
+import { GitAltIcon } from '@patternfly/react-icons';
+import GeneratorView from './GeneratorView';
+
+interface GitGeneratorProps {
+  generator: any;
+}
+
+const GitGenerator: React.FC<GitGeneratorProps> = ({ generator }) => {
+  const generatorType = generator.files ? "File" : "Directory";
+  
+  return (
+    <GeneratorView icon={<GitAltIcon />} title={`git (${generatorType})`}>
+      <DescriptionList isHorizontal isCompact>
+        {generator.repoURL && (
+          <DescriptionListGroup>
+            <DescriptionListTerm>Repository</DescriptionListTerm>
+            <DescriptionListDescription>{generator.repoURL}</DescriptionListDescription>
+          </DescriptionListGroup>
+        )}
+        {generator.revision && (
+          <DescriptionListGroup>
+            <DescriptionListTerm>Revision</DescriptionListTerm>
+            <DescriptionListDescription>{generator.revision}</DescriptionListDescription>
+          </DescriptionListGroup>
+        )}
+        {generator.directories && (
+          <DescriptionListGroup>
+            <DescriptionListTerm>Directories</DescriptionListTerm>
+            <DescriptionListDescription>{generator.directories.length} directory(ies)</DescriptionListDescription>
```

Comment:
We should explicitly say: `1 directory`
and for more than 1, say: `2 directories`
It will be more polished that way